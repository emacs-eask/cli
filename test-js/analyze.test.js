const { TestContext } = require("./helpers");

describe("analyze", () => {
  describe("in ./dsl", () => {
    const ctx = new TestContext("./test-js/dsl");

    it("handles plain text", async () => {
      await ctx.runEask("analyze");
      await ctx.runEask("analyze Eask");
    });

    it("handles json option", async () => {
      const { stderr } = await ctx.runEask("analyze --json");
      await ctx.runEask("analyze Eask --json");

      // try to parse output, errors if not valid
      JSON.parse(stderr);
    });

    it("matches snapshot", async () => {
      const res = await ctx.runEask("analyze");
      const resClean = res.sanitized().raw();
      expect(resClean).toMatchSnapshot();
    });

    it("should report multiple definitions", async () => {
      const { stderr } = await ctx.runEask("analyze");
      // expect this substring
      expect(stderr).toMatch("Multiple definition of `package'");
    });
  });

  describe("in ./empty", () => {
    const ctx = new TestContext("./test-js/empty");

    it("should error", async () => {
      await expect(ctx.runEask("analyze")).rejects.toThrow();
    });
  });

  describe("in ./metadata", () => {
    const ctx = new TestContext("./test-js/metadata");

    it("handles plain text", async () => {
      await ctx.runEask("analyze");
      await ctx.runEask("analyze Eask");
    });

    it("handles json", async () => {
      await ctx.runEask("analyze --json");
      await ctx.runEask("analyze Eask --json");
    });
  });
});
