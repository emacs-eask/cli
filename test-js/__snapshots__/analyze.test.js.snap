// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyze in ./dsl matches snapshot 1`] = `
{
  "stderr": "
~/Eask:7:18 Error: Multiple definition of \`package'
~/Eask:10:55 Error: Multiple definition of \`website-url'
~/Eask:12:16 Error: Multiple definition of \`keywords'
~/Eask:15:15 Warning: Warning regarding duplicate author name, name
~/Eask:18:17 Warning: Warning regarding duplicate license name, GPLv3
~/Eask:21:29 Error: Multiple definition of \`package-file'
~/Eask:23:35 Warning: Pkg-file seems to be missing \`check-pkg.el'
~/Eask:24:35 Error: Multiple definition of \`package-descriptor'
~/Eask:28:61 Error: Run-script with the same key name is not allowed: \`test\`
~/Eask:31:13 Error: Multiple definition of source \`gnu'
~/Eask:33:24 Error: Unknown package archive \`magic-archive'
~/Eask:36:20 Error: Define dependencies with the same name \`emacs'
~/Eask:39:19 Error: Define dependencies with the same name \`dash'
~/Eask:44:2 Error: Define dependencies with the same name \`f'
",
  "stdout": "",
}
`;
